volume/volume.go

```diff
@@ -194,6 +195,18 @@ type BlockDriver interface {
 	Detach(volumeID string, options map[string]string) error
 }
 
+// CredsDriver provides methods to handle credentials
+type CredsDriver interface {
+	// CredsCreate creates credential for a given cloud provider
+	CredsCreate(params map[string]string) (string, error)
+	// CredsList lists the configured credentials in the cluster
+	CredsList() (map[string]interface{}, error)
+	// CredsDelete deletes the credential associated schedUUID
+	CredsDelete(schedUUID string) error
```

Comment:
creds interface should not deal with *sched* . Interface should just be Create/Delete/Enumerate

Reply:
Good catch with naming!. Will fix it.